internal/uvm/mapped_directory.go

```diff
@@ -0,0 +1,25 @@
+package uvm
+
+import (
+	"context"
+
+	"github.com/Microsoft/hcsshim/internal/guestrequest"
+	"github.com/Microsoft/hcsshim/internal/requesttype"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+// MappedDirectory issues a mapped directory guest request.
+func (uvm *UtilityVM) MappedDirectory(ctx context.Context, hostPath, containerPath string, readOnly bool) error {
```

Comment:
I think it would be better to just expose a guest request API at this layer, and require the caller to build up the request themselves. We already have a lot of code that creates guest requests anyways.

Reply:
I had this originally but I changed it as I thought it was a little odd. You mean essentially just 

```go
func (uvm *UtilityVM) GuestRequest(ctx context.Context, guestReq interface{}) {
       msr := &hcsschema.ModifySettingRequest{
		GuestRequest: guestreq,
	}
	return uvm.modify(ctx, msr)
}
```

Reply:
Yeah, that will let us change all of the places we do guest requests over to use this API, and then we can move it out to a separate interface (or break it into separate calls like your `MappedDirectory` example) later.

Reply:
Yea that works. I'll change it to use this for now and we can do cleanup later.